tests/suites/hup/suite.js

```diff
@@ -12,14 +12,35 @@ const { join } = require('path');
 const { homedir } = require('os');
 const Docker = require('dockerode');
 const exec = require('bluebird').promisify(require('child_process').exec);
+const Bluebird = require('bluebird');
 
 // Starts registry, uploads target image to registry
 const runRegistry = async (that, seedWithImage) => {
 	const docker = new Docker();
-	const registryImage = 'registry:2';
-
-	that.log('Pulling registry image');
-	await docker.pull(registryImage);
+	const registryImage = 'registry:2.7.1';
+
+	that.log(`Pulling registry image: ${registryImage}`);
+	const stream = await docker.pull(registryImage);
+
+	await new Bluebird((resolve, reject) => {
```

Comment:
An alternative to Bluebird here would be to install the stream-to-promise package.

Reply:
Before I forget, I solve a similar problem using promises worth a shot, no imports required.

Reply:
Thanks for the tip @vipulgupta2048! How does this look?
```nodejs
        that.log(`Pulling registry image: ${registryImage}`);
	await new Promise(async (resolve, reject) => {
		await docker.pull(registryImage, function(err, stream) {
			if (err) reject(err);
			stream.once('end', resolve);
		});
	});
```